### PR TITLE
[RFR] Add `linkType` to ReferenceField

### LINKED ISSUE #247: Disable 'link to edit' in ReferenceField
Hi,

First, I always to say thank every time I'm here.
Please stop by to see what you've done to us :)
nab-sale-management.herokuapp.com (user/pass: admin/changeit)

I see in source code that we use <Link/> in ReferenceField and click on it will browse us to edit page of referenced items.

It's great thing to implement, but can we add property to enable/disable it?
When disable, it should be rendered as <span/> or as same as other inline fields.
First, it can prevent user edit reference fields
Second, if there are many reference fields, there are also many blue links/hrefs in grid.

I can override this component to satisfy my need but I think it's pretty nice to improve in next release.

Thank you very much.

### PR DESCRIPTION
**Pull request by fzaninotto**
Supersedes #486 by fixing coding standards and documentation, and adding tests.

Fixes #247.

### CODE REVIEW

**Comment by leesei** [comment]
If we are to support boolean `linkType`, should line 65 change to 
```js
if (linkType === 'edit' || linkType === true) {
```

**Comment by fzaninotto** [comment]
@leesei nice to have, but good enough as is.

File: src/mui/field/ReferenceField.js [hunk 1]
```diff
@@ -4,12 +4,35 @@ import { Link } from 'react-router';
...
+ *     <TextField source="name" />
+ * </ReferenceField>
+ *
+ * By default, includes a link to the <Edit> page of the related record
+ * (`/users/:userId` in the previous example).
+ *
+ * Set the linkType prop to "show" to link to the <Show> page instead.
+ *
+ * @example
+ * <ReferenceField label="User" source="userId" reference="users" linkType="edit">
```
Comment (djhi) [inline src/mui/field/ReferenceField.js]: s/edit/show/

**Comment by fzaninotto** [comment]
Code review applied

**Comment by michaeI** [comment]
What about referenceArrayField, how to disable link there?

**Pull request merged**